app/src/main/java/com/example/georgesamuel/dubaihotels/presentation/features/items/ListItemsActivity.java

```diff
@@ -0,0 +1,80 @@
+package com.example.georgesamuel.dubaihotels.presentation.features.items;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.constraintlayout.widget.ConstraintLayout;
+import androidx.lifecycle.ViewModelProviders;
+import androidx.recyclerview.widget.RecyclerView;
+import androidx.recyclerview.widget.StaggeredGridLayoutManager;
+
+import android.app.ProgressDialog;
+import android.os.Bundle;
+import android.util.Log;
+
+import com.example.georgesamuel.dubaihotels.R;
+import com.example.georgesamuel.dubaihotels.entities.Hotel;
+import com.google.android.material.snackbar.Snackbar;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+public class ListItemsActivity extends AppCompatActivity {
+
+
+    @BindView(R.id.hotel_Items_recycler_view)
+    RecyclerView hotelItemsRecyclerView;
+
+    @BindView(R.id.parent_constraint_layout)
+    ConstraintLayout parentConstraintLayout;
+
+    private List<Hotel> list;
+    private ProgressDialog progressDialog;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        ButterKnife.bind(this);
+        progressDialog = new ProgressDialog(this);
+        initRecycler();
+
+        ListItemsViewModel viewModel = ViewModelProviders.of(this).get(ListItemsViewModel.class);
+        viewModel.getDetails();
+        viewModel.result.observe(this, detailsModel -> {
```

Comment:
divide code to small methods like observeGetHotels(),observeLoadingIndecator()

Reply:
ok , i devided them